pctasks/run/tests/workflow/test_kubernetes.py

```diff
@@ -0,0 +1,305 @@
+                "options": {"skip_validation": False},
+                "cosmos_endpoint": cosmos,
+                "db_name": "lowlatencydb",
+                "create_items_function": function,
+                "collection_id": "test",
+            },
+            "schema_version": "1.0.0",
+        }
+    )
+
+
+def test_get_deployment_name(task_definition):
+    result = pctasks.run.workflow.kubernetes.get_deployment_name(task_definition)
+    assert result == "devstoreaccount1-test-deployment"
+
+
+def test_build_streaming_deployment(task_definition):
+    result = pctasks.run.workflow.kubernetes.build_streaming_deployment(
+        task_definition,
+        input_uri="blob://pctasksteststaging/input",
+        taskio_tenant_id="test-tenant-id",
+        taskio_client_id="test-client-id",
+        taskio_client_secret="test-client-secret",
+    )
+    labels = {"node_group": "pc-lowlatency"}
+    assert result.metadata.name == "devstoreaccount1-test-deployment"
+    assert result.metadata.labels == labels
+    assert result.spec.selector["matchLabels"] == labels
+    assert result.spec.template.metadata.labels == labels
+    assert result.spec.template.spec.service_account_name == "default"
+    assert result.spec.template.spec.containers[0].image == task_definition.image
+    assert result.spec.template.spec.containers[0].args == [
+        "task",
+        "run",
+        "blob://pctasksteststaging/input",
+    ]
+
+
+def test_build_streaming_scaler(task_definition):
+    result = pctasks.run.workflow.kubernetes.build_streaming_scaler(task_definition)
+    assert result["metadata"]["name"] == "devstoreaccount1-test-scaler"
+    assert (
+        result["spec"]["scaleTargetRef"]["name"] == "devstoreaccount1-test-deployment"
+    )
+    assert result["spec"]["minReplicaCount"] == 0
+    assert result["spec"]["maxReplicaCount"] == 10
+    assert result["spec"]["pollingInterval"] == 30
+    assert result["spec"]["triggers"][0]["metadata"]["queueName"] == "test"
+    assert result["spec"]["triggers"][0]["metadata"]["queueLength"] == "100"
+    assert (
+        result["spec"]["triggers"][0]["metadata"]["accountName"] == "devstoreaccount1"
+    )
+
+
+@pytest.fixture(scope="session")
+def run_settings():
+    """Run settings, using azurite for storage."""
+    key = (
+        "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq"
+        "/K1SZFPTOtr/KBHBeksoGMGw=="
+    )
+    return RunSettings(
+        notification_queue={
+            "account_url": "queue://devstoreaccount1/notifications",
+            "connection_string": "connstr",
+            "queue_name": "notifications",
+            "sas_token": "sas",
+        },
+        tables_account_url="http://127.0.0.1:10001",
+        tables_account_name="devstoreaccount1",
+        tables_account_key="devstoreaccount1",
+        blob_account_url="http://127.0.0.1:10000",
+        blob_account_name="devstoreaccount1",
+        blob_account_key=key,
```

Comment:
I think this configuration isn't quite right for the containerized test setup. It's working for me locally where I can access azurite at this URL. But I think inside docker it's available at `azurite:10000`